hs_app_netCDF/receivers.py

```diff
@@ -184,19 +184,24 @@ def netcdf_pre_delete_file_from_resource(sender, **kwargs):
     del_file = kwargs['file']
     del_file_ext = os.path.splitext(del_file.resource_file.name)[-1]
 
-    # TODO: update resource modification info
-    # user = kwargs['user']
-    # resource_modified(resource, user)
+    # update resource modification info
+    user = nc_res.creator
+    utils.resource_modified(nc_res, user)
 
     # delete the netcdf header file or .nc file
-    file_ext = ['.nc', '.txt']
+    file_ext = {'.nc': 'application/x-netcdf',
+                '.txt': 'text/plain'}
+
     if del_file_ext in file_ext:
-        file_ext.remove(del_file_ext)
+        del file_ext[del_file_ext]
         for f in ResourceFile.objects.filter(object_id=nc_res.id):
             ext = os.path.splitext(f.resource_file.name)[-1]
             if ext in file_ext:
                 f.resource_file.delete()
                 f.delete()
+                format_element = nc_res.metadata.formats.filter(value=file_ext[ext]).first()
+                if format_element:
+                    nc_res.metadata.delete_element(format_element.term, format_element.id)
```

Comment:
Do you ever expect there to be more than one and if so why do you only want to delete the first? If you only exist one, why not just do `delete()` instead of `first()` on the `QuerySet`?